app/org/sagebionetworks/bridge/models/schedules/TaskScheduler.java

```diff
@@ -51,36 +42,43 @@ protected DateTime getScheduledTimeBasedOnEvent(Schedule schedule, Map<String, D
         return eventTime;
     }
     
-    protected void addTaskForEachTime(List<Task> tasks, DateTime scheduledTime) {
+    protected void addTaskForEachTime(List<Task> tasks, ScheduleContext context, DateTime scheduledTime) {
         if (schedule.getTimes().isEmpty()) {
             // We're using whatever hour/minute/seconds were in the original event.
-            addTaskForEachActivityAtTime(tasks, scheduledTime);
+            addTaskForEachActivityAtTime(tasks, context, scheduledTime);
         } else {
             for (LocalTime time : schedule.getTimes()) {
-                scheduledTime = new DateTime(scheduledTime).withTime(time);
-                addTaskForEachActivityAtTime(tasks, scheduledTime);
+                scheduledTime = new DateTime(scheduledTime, context.getZone()).withTime(time);
+                addTaskForEachActivityAtTime(tasks, context, scheduledTime);
             }
         }
     }
     
-    private void addTaskForEachActivityAtTime(List<Task> tasks, DateTime scheduledTime) {
+    private void addTaskForEachActivityAtTime(List<Task> tasks, ScheduleContext context, DateTime scheduledTime) {
+        // Assert that the scheduledTime was constructed by subclass implementation with the correct time zone.
+        checkArgument(context.getZone().equals(scheduledTime.getZone()), 
+            "Scheduled DateTime does not have requested time zone: " + scheduledTime.getZone());
+
         // If this time point is outside of the schedule's active window, skip it.
-        if (isInWindow(schedule, scheduledTime)) {
+        if (isInWindow(scheduledTime)) {
             // As long at the tasks are not already expired, add them.
-            DateTime expiresOn = getExpiresOn(scheduledTime, schedule);
-            if (expiresOn == null || expiresOn.isAfter(now)) {
+            DateTime expiresOn = getExpiresOn(scheduledTime);
+            if (expiresOn == null || expiresOn.isAfter(context.getNow())) {
                 for (Activity activity : schedule.getActivities()) {
-                    DynamoTask task = new DynamoTask();
-                    task.setSchedulePlanGuid(schedulePlanGuid);
+                    Task task = new DynamoTask();
```

Comment:
Task.create() has been done and checked in. Ultimately, we're exposing how we localize this task which as mentioned in a longer email, I would have preferred to avoid. It's also as you say, minor picky stuff.